jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/Button.java

```diff
@@ -3,12 +3,66 @@
 import com.epam.jdi.light.common.JDIAction;
 import com.epam.jdi.light.elements.base.UIBaseElement;
 import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.complex.ISetup;
+import com.epam.jdi.light.material.annotations.JDIButtons;
 import com.epam.jdi.light.material.asserts.inputs.ButtonAssert;
 
-public class Button extends UIBaseElement<ButtonAssert> {
+import java.lang.reflect.Field;
 
+import static com.epam.jdi.light.elements.pageobjects.annotations.objects.FillFromAnnotationRules.fieldHasAnnotation;
+
+public class Button extends UIBaseElement<ButtonAssert> implements ISetup {
     Button() {}
 
+    String containedButton;
+    String textButton;
+    String iconLabelButton;
+    String iconLabelIcon;
+    String iconLabelSpanIcon;
+    String iconButton;
+    String customizedButton;
+    String complexButton;
+
+    @JDIAction("Get contained button by index")
+    public Button getContainedButtonByIndex(int index) {
+        return new Button(core().finds(containedButton).get(index));
+    }
+
+    @JDIAction("Get text button by index")
+    public Button getTextButtonByIndex(int index) {
+        return new Button(core().finds(textButton).get(index));
+    }
+
+    @JDIAction("Get icon label button by index")
+    public Button getIconLabelButtonByIndex(int index) {
+        return new Button(core().finds(iconLabelButton).get(index));
+    }
+
+    @JDIAction("Get icon label icon by index")
+    public Button getIconLabelIconByIndex(int index) {
+        return new Button(core().finds(iconLabelIcon).get(index));
+    }
+
+    @JDIAction("Get icon label span icon")
+    public Button getIconLabelSpanIcon() {
+        return new Button(core().find(iconLabelSpanIcon));
+    }
+
+    @JDIAction("Get icon button by index")
+    public Button getIconButtonByIndex(int index) {
+        return new Button(core().finds(iconButton).get(index));
+    }
+
+    @JDIAction("Get customized button by index")
+    public Button getCustomizedButtonByIndex(int index) {
+        return new Button(core().finds(customizedButton).get(index));
+    }
+
+    @JDIAction("Get complex button by index")
+    public Button getComplexButtonByIndex(int index) {
+        return new Button(core().finds(complexButton).get(index));
+    }
+
```

Comment:
Please use 1 getButtonByGroupIndex(String group, int index) instead of  for complex and other variants. group = e.g. "Complex"